packages/driver/src/cy/commands/cookies.ts

```diff
@@ -336,61 +317,61 @@ export default function (Commands, Cypress, cy, state, config) {
       // @see https://web.dev/samesite-cookies-explained/#changes-to-the-default-behavior-without-samesite
       if (cookie.sameSite === 'no_restriction' && cookie.secure !== true) {
         $errUtils.throwErrByPath('setCookie.secure_not_set_with_samesite_none', {
-          onFail,
+          onFail: log,
           args: {
             value: options.sameSite, // for clarity, throw the error with the user's unnormalized option
           },
         })
       }
 
       if (!_.isString(name) || !_.isString(value)) {
-        $errUtils.throwErrByPath('setCookie.invalid_arguments', { onFail })
+        $errUtils.throwErrByPath('setCookie.invalid_arguments', { onFail: log })
       }
 
-      if (options.name.startsWith('__Secure-') && cookieValidatesSecurePrefix(options)) {
-        $errUtils.throwErrByPath('setCookie.secure_prefix', { onFail })
+      if (name.startsWith('__Secure-') && cookieValidatesSecurePrefix(options)) {
+        $errUtils.throwErrByPath('setCookie.secure_prefix', { onFail: log })
       }
 
-      if (options.name.startsWith('__Host-') && cookieValidatesHostPrefix(options)) {
-        $errUtils.throwErrByPath('setCookie.host_prefix', { onFail })
+      if (name.startsWith('__Host-') && cookieValidatesHostPrefix(options)) {
+        $errUtils.throwErrByPath('setCookie.host_prefix', { onFail: log })
       }
 
+      validateDomainOption(options.domain, 'setCookie', log)
+
       Cypress.emit('set:cookie', cookie)
 
       return cy.retryIfCommandAUTOriginMismatch(() => {
-        return automateCookies('set:cookie', cookie, options._log, responseTimeout)
+        return automateCookies('set:cookie', cookie, log, responseTimeout)
         .then(pickCookieProps)
-        .then((resp) => {
-          options.cookie = resp
-
-          return resp
-        }).catch(handleBackendError('setCookie', 'setting the requested cookie in', onFail))
+        .tap((result) => {
+          resultingCookie = result
+        }).catch(handleBackendError('setCookie', 'setting the requested cookie in', log))
       }, options.timeout)
     },
 
-    clearCookie (name, userOptions: Partial<Cypress.Loggable & Cypress.Timeoutable> = {}) {
-      const options: InternalClearCookieOptions = _.defaults({}, userOptions, {
+    clearCookie (name, userOptions: Partial<Cypress.CookieOptions> = {}) {
```

Comment:
I think my above suggestion to make domain option would remove the need for `Partial<Cypress.CookieOptions>` throughout since each option should be optional

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/24471/commits/d4e600dd589e1a6ff6b6bfcf25ca1d193a3c8fc0